src/components/user-info.js

```diff
@@ -3,21 +3,40 @@ import React from 'react';
 const INFO = 'info';
 
 export default function Info (props) {
-    const { info = {}, onChange } = props;
-    const _onChange               = e => onChange(e, INFO);
+    function errorClass (valid) {
+        return valid ? '' : 'field-error';
```

Comment:
Let's use a light weight package for this purpose 
https://www.npmjs.com/package/classnames